core/application/impl/kagome_application_impl.cpp

```diff
@@ -35,47 +37,21 @@ namespace kagome::application {
     jrpc_api_service_ = injector_.create<sptr<api::ApiService>>();
   }
 
-  // TODO (yuraz) rewrite when there will be more info
-  Epoch KagomeApplicationImpl::makeInitialEpoch() {
-    std::vector<primitives::Authority> authorities;
-    auto &&session_keys = config_storage_->getSessionKeys();
-    authorities.reserve(session_keys.size());
-
-    for (auto &item : session_keys) {
-      authorities.push_back(primitives::Authority{{item}, 1u});
-    }
-
-    // some threshold value resulting in producing block in every slot. In
-    // future will be calculated using runtime
-    boost::multiprecision::uint128_t threshold(
-        "102084710076281554150585127412395147264");
-
-    Randomness rnd{};
-    rnd.fill(0u);
-
-    Epoch value{.epoch_index = 0u,
-                .start_slot = 0u,
-                .epoch_duration = 100,
-                .slot_duration = 1000ms,
-                .authorities = authorities,
-                .threshold = threshold,
-                .randomness = rnd};
-
-    return value;
-  }
-
   void KagomeApplicationImpl::run() {
     jrpc_api_service_->start();
-    auto epoch = makeInitialEpoch();
-    babe_->runEpoch(std::move(epoch), clock_->now());
-
+    if (is_genesis_epoch_) {
+      babe_->runGenesisEpoch();
+    }
     grandpa_launcher_->start();
```

Comment:
How would be run BABE for no genesis epoch?

Reply:
AFAIU, after calling runGenesisEpoch BABE will continue running on its own. However, I'm not sure if this is reflected in BABE docs, and if it's not it's better to add it there.